fluss-client/src/main/java/com/alibaba/fluss/client/admin/FlussAdmin.java

```diff
@@ -323,11 +330,21 @@ private static void sendListOffsetsRequest(
                                         for (PbListOffsetsRespForBucket resp :
                                                 r.getBucketsRespsList()) {
                                             if (resp.hasErrorCode()) {
-                                                bucketToOffsetMap
-                                                        .get(resp.getBucketId())
-                                                        .completeExceptionally(
-                                                                ApiError.fromErrorMessage(resp)
-                                                                        .exception());
+                                                ApiException exception =
+                                                        ApiError.fromErrorMessage(resp).exception();
+                                                if (exception
+                                                        instanceof InvalidTimestampException) {
```

Comment:
It's not encouraged to use exceptions as common information exchanging. I would recommend this can be handled in server-side to simplify client-side implementation (and future other clients).